src/persistence/StorageNode.ts

```diff
@@ -0,0 +1,102 @@
+/**
+ * @license
+ * Copyright 2022 The node-matter Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Time } from "../time/Time";
+import { Storage } from "./Storage";
+import { readFile, writeFile } from "fs/promises";
+import { Logger } from "../log/Logger";
+import { ByteArray } from "@project-chip/matter.js";
+import {StorageInMemory} from "./StorageInMemory";
+
+/** We store changes 1s after a value was set to the storage, but not more often than every 1s. */
+const COMMIT_DELAY = 1000 /* 1s */;
+
+const logger = Logger.get("StorageNode");
+
+export class StorageNode extends StorageInMemory {
+    private readonly commitTimer = Time.getTimer(COMMIT_DELAY, () => this.commit());
+    private waitForCommit = false;
+    private allowToCommit = false;
+
+    constructor(
+        private readonly path: string,
+    ) {
+        super();
+    }
+
+    async initialize() {
+        if (Object.keys(this.store).length > 0) {
+            throw new Error("Storage contains already values, can not initialize!");
+        }
+        try {
+            this.store = this.fromJson(await readFile(this.path, "utf-8"));
+        } catch (error: any) {
+            if (error.code !== "ENOENT") {
+                throw error;
+            }
+        }
+        this.allowToCommit = true;
+    }
+
+    set<T>(context: string, key: string, value: T): void {
+        super.set(context, key, value);
+        if (!this.waitForCommit) {
+            this.waitForCommit = true;
+            this.commitTimer.start();
+        }
+    }
+
+    private async commit() {
+        if (!this.allowToCommit) return;
+        this.waitForCommit = false;
+        try {
+            await writeFile(this.path, this.toJson(this.store), "utf-8");
+        } catch (error) {
+            logger.error("Failed to write storage file", error);
```

Comment:
Do we really want to catch and continue this error?
Might be fine to just let it crash the device, since things are not working as expected.

Reply:
The problem is that this error would be not catchable at all because triggered from within a timer, but Ok lets do that for now and see if we get issues with it